test/Account/getNonce.test.ts

```diff
@@ -0,0 +1,55 @@
+import sinon from 'sinon';
+
+import BigNumber from 'bignumber.js';
+import Account from '../../src/Account';
+import assert from '../test_utils/assert';
+import SpyAssert from '../test_utils/SpyAssert';
+
+describe('Account.getNonce', (): void => {
+  let dummyInitialNonce: BigNumber;
+  let web3MockObject: any;
+  const accountObject: Account = new Account(
+    '0x0000000000000000000000000000000000000001',
+  );
+
+  beforeEach((): void => {
+    dummyInitialNonce = new BigNumber('1');
+    web3MockObject = {
+      eth: {
+        getTransactionCount: sinon.fake.resolves(dummyInitialNonce),
+      },
+    };
+  });
+
+  afterEach(async (): Promise<void> => {
+    sinon.restore();
+  });
+
+  it('should return nonce by calling getTransactionCount on web3', async (): Promise<void> => {
+    const nonce: BigNumber = await accountObject.getNonce(web3MockObject);
+    assert.notStrictEqual(
+      nonce,
+      dummyInitialNonce,
+      'nonce should match',
+    );
+    SpyAssert.assert(
+      web3MockObject.eth.getTransactionCount,
+      1,
+      [[accountObject.address, 'pending']],
+    );
+  });
+
+  it('should return incremented nonce from in-memory storage', async (): Promise<void> => {
+    const nonce: BigNumber = await accountObject.getNonce(web3MockObject);
```

Comment:
I feel  accountObject should go to beforeEach. This will make sure unit tests are atomic. Else 2nd unit unit test is dependent on 1st unit test for persistence of memory variable.

We can call getNonce 2 times in the 2nd unit test.

Reply:
done.